### PR TITLE
feat(plugin): StatusWhilePlaying

### PR DESCRIPTION
**Pull request by thororen1234**
This plugin automatically updates your status when playing games and reverts the status back when the game is closed.

### CODE REVIEW

**Comment by sadan4** [comment]
this looks too selfbot-y 
also RPC

**Comment by Vendicated** [comment]
won't work on web or vesktop

**Comment by thororen1234** [comment]
my fault ill rename it rn

File: src/plugins/friendsSince/index.tsx
Comment (Scyye) [inline src/plugins/friendsSince/index.tsx:1]: Why is this here?

Comment (thororen1234) [reply src/plugins/friendsSince/index.tsx:1]: because I didn't sync anything yet (this was a force push commit from vee)

Comment (Scyye) [reply src/plugins/friendsSince/index.tsx:1]: gotta love force pushes. 🤦

File: src/plugins/statusWhilePlaying.desktop/index.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        } else if (event.games.length === 0) {
+            updateAsync(savedStatus);
+        }
+    },
+    start() {
+        FluxDispatcher.subscribe("RUNNING_GAMES_CHANGE", this.runningGamesChange);
+    },
+    stop() {
+        FluxDispatcher.unsubscribe("RUNNING_GAMES_CHANGE", this.runningGamesChange);
+    }
```
Comment (Vendicated) [inline src/plugins/statusWhilePlaying.desktop/index.ts:61]:
```suggestion
    flux: {
    	RUNNING_GAMES_CHANGE(event) {
	        let savedStatus = "";
	        if (event.games.length > 0) {
	            const status = PresenceStore.getStatus(UserStore.getCurrentUser().id);
	            savedStatus = status;
	            updateAsync(settings.store.statusToSet);
	        } else if (event.games.length === 0) {
	            updateAsync(savedStatus);
	        }
    	},
    }
```

File: src/plugins/statusWhilePlaying.desktop/index.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    authors: [Devs.thororen],
+    settings,
+    runningGamesChange(event) {
+        let savedStatus = "";
+        if (event.games.length > 0) {
+            const status = PresenceStore.getStatus(UserStore.getCurrentUser().id);
+            savedStatus = status;
+            updateAsync(settings.store.statusToSet);
+        } else if (event.games.length === 0) {
+            updateAsync(savedStatus);
```
Comment (Vendicated) [inline src/plugins/statusWhilePlaying.desktop/index.ts:53]: This doesn't make sense. savedStatus is always empty string here. Did you mean to declare `savedStatus` outside of the function?

File: src/plugins/statusWhilePlaying.desktop/index.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                label: "Invisible",
+                value: "invisible",
+            }
+        ]
+    }
+});
+
+export default definePlugin({
+    name: "StatusWhilePlaying",
+    description: "Automatically updates your status when playing games",
```
Comment (Vendicated) [inline src/plugins/statusWhilePlaying.desktop/index.ts:43]:
```suggestion
    description: "Automatically updates your online status (online, idle, dnd) when launching games",
```

File: src/plugins/statusWhilePlaying.desktop/index.ts (hunk 2 above)
Comment (thororen1234) [reply src/plugins/statusWhilePlaying.desktop/index.ts]: In my testing on game launch it does set saved status to the prior status and then on game close it sets it to the status prior to launching the game.
However it might be best to declare it outside the function what are your thoughts?

Comment (thororen1234) [reply src/plugins/statusWhilePlaying.desktop/index.ts]: Nvm I have no clue why I did that such a weird way. savedStatus doesn't exist anymore and now just uses status in place of it, no redefines nothing.

**Comment by thororen1234** [comment]
Alright so it now checks if the status being set and the current status are the same. For the savedStatus it will get a value when a game is launched (game length is above 0) and used when the game is closed ( game length is 0) I have also moved it outside of the flux event and it also checks if it isn't "" when updating to the old status.

**Comment by thororen1234** [comment]
Moved to [here](https://github.com/Vendicated/Vencord/pull/2988).

**Comment by Scyye** [comment]
why?
